### ISSUE TITLE
`options` getting called on a `skip`

### ISSUE THREAD

**Issue by jbaxleyiii** [apollographql/apollo-link-state#23]
**Intended outcome:**

For illustration, use this HOC:

```es6
const withDependency = graphql(DependencyQuery,
  {
    skip: ({dependency}) => !dependency || !dependency.id,
    options: ({dependency}) => ({
      variables: {
        id: dependency.id
      }
    })
  }
)
```

I expect options to not be evaluated as a result of the `skip` test failing. 

**Actual outcome:**
`options` is called when `dependency` doesn't exist, resulting in a javascript prop access error.

**Version**

* apollo-client@1.4.16
* react-apollo@1.9.3

**Comment by jbaxleyiii** [comment]
➤ Simon Tegg commented:

This is a duplicate of #903 

There's a fix in v.2 but I'm unable to upgrade as a I depend on redux

**Comment by jbaxleyiii** [comment]
➤ David Glasser commented:

This was fixed in #1181 but that didn't make it to 2.0 somehow. #1330 should resolve that.
